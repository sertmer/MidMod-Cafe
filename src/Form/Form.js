import React, { Component } from 'react'

class Form extends Component {
  constructor() {
    super()

    this.state = {
      title: '',
      description: ''
    }
  }

  render() {
    return (
      <>
      <input 
        placeholder=''
        name='title'
      />
      </>
    )
  }
}

export default Form