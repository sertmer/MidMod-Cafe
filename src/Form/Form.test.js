import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper
  let mockEvent
  let mockDefaultState
  let mockUpdatedState
  let mockAddReservation
  let mockHandleChange
  beforeEach(() => {
    mockEvent = { target: { name: 'name', value: 'bob' } }
  
    mockDefaultState = {
      name: '',
      date: '',
      time: '',
      number: 1
    }
  
    mockUpdatedState = {
      name: 'bob',
      date: '',
      time: '',
      number: 1
    }

    mockAddReservation = jest.fn()

    wrapper = shallow(<Form addReservation={mockAddReservation}/>)

  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state when handleChange is invoked', () => {
    
    expect(wrapper.state()).toEqual(mockDefaultState)

    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state()).toEqual(mockUpdatedState)
  })

  it('should call addReservation on button click', () => {
    wrapper.find('button').simulate('click', mockEvent)
    expect(wrapper.instance().props.addReservation).toHaveBeenCalledWith(mockDefaultState)
  })

  it('should call handleChange when it detects a change on a form', () => {
    wrapper.instance().handleChange = jest.fn()
    wrapper.instance().forceUpdate()
    wrapper.find('.name-input').simulate('change', mockEvent)
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
  })
})