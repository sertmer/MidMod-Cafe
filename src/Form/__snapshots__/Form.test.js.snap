// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<section>
  <input
    className="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
    value={1}
  />
  <button
    onClick={[Function]}
  >
    Submit, why dontcha?
  </button>
</section>
`;
